hal/halx86/up/pic.c

```diff
@@ -405,7 +410,7 @@ PHAL_SW_INTERRUPT_HANDLER SWInterruptHandlerTable[20] =
 /* Handlers for pending software interrupts when we already have a trap frame*/
 PHAL_SW_INTERRUPT_HANDLER_2ND_ENTRY SWInterruptHandlerTable2[3] =
 {
-    (PHAL_SW_INTERRUPT_HANDLER_2ND_ENTRY)KiUnexpectedInterrupt,
+    (PHAL_SW_INTERRUPT_HANDLER_2ND_ENTRY)KiUnexpectedInterrupt2ndEntry,
```

Comment:
A wrapper function seems excessive for fixing a broken warning. Does a `(PVOID)` cast make the warning go away?

Reply:
Yes, `(PHAL_SW_INTERRUPT_HANDLER_2ND_ENTRY)(PVOID)` or `(PVOID)` would work too.
Though I understand your "minimal change" approach, that one feels like an obscure "hack" to me.
I feel the warning is "correct" and I prefer an explicit prototype (both calling convention and argument).
Choose one of the three...

Reply:
I prefer `(PHAL_SW_INTERRUPT_HANDLER_2ND_ENTRY)(PVOID)` to stay in line with the array above.
If the function was anything other than a zero-parameter function that immediately bugchecks, I'd be in favor of a wrapper. But since it is what it is, that just adds noise.

Reply:
Done, plus a comment.